client/src/Components/ResultsTable.js

```diff
@@ -0,0 +1,114 @@
+import React from "react";
+import { useTable } from "react-table";
+const [results,setAllResults]=useState();
+const Styles = styled.div`
+  padding: 1rem;
+
+  table {
+    border-spacing: 0;
+    border: 1px solid black;
+
+    tr {
+      :last-child {
+        td {
+          border-bottom: 0;
+        }
+      }
+    }
+
+    th,
+    td {
+      margin: 0;
+      padding: 0.5rem;
+      border-bottom: 1px solid black;
+      border-right: 1px solid black;
+
+      :last-child {
+        border-right: 0;
+      }
+    }
+  }
+`;
+export default function ResultsTable() {
+	const [studentNames, setStudentNames]=useState([]);
+	const [selQuizQuestions, setSelQuizQuestions]=useState([]);
+	const [answeredQuestons, setAnsweredQuestons]=useState([]);
+	const [allResults, setAllResults]=useState(null);
+	const [quizSelected, setQuizSelected]=useState(null);
+	const fetchResults =()=>{
+		fetch("http://localhost:3100/api/results/")
+			.then((response)=>response.json())
+			.then((data)=>{
+				setAllResults(data);
+				console.log({ data });
+			});
+	};
+	useEffect(() => {
+		fetchResults();
+	},[]);
```

Comment:
you don't need this here, you can just get it by props

Reply:
Getting by props will be incorrect as other components do not need results. As a rule it is better to keep fetched data states closer to where they are used.. In other components we fetch in app because more than 1 child use them, here it is not the case

Reply:
there will be [route] at the end of use effect that you asked for in the beguinning